### PR TITLE
Disable transparent huge pages

### PR DESCRIPTION
**Pull request by msaffitz**
Following best practices from http://redis.io/topics/latency

### CODE REVIEW

**Comment by ifeltsweet** [comment]
+1

File: providers/configure.rb [hunk 1]
```diff
@@ -19,10 +19,19 @@
...
+  sysconfig
   configure
   new_resource.updated_by_last_action(true)
 end
 
+def sysconfig
+  hugepage_file = "/sys/kernel/mm/transparent_hugepage/enabled"
+  file hugepage_file do
+    content 'never'
+    only_if { ::File.exists?(hugepage_file) }
```
Comment (shortdudey123) [inline providers/configure.rb]: Can you add another check to verify the contents of the file? 
`&& File.read(hugepage_file).chomp == 'never'`

That way it gets set back to never if it gets changed accidently

Comment (josephholsten) [inline providers/configure.rb]:
```suggestion
    only_if { ::File.exists?(hugepage_file) && File.read(hugepage_file).chomp == 'never' }
```

**Comment by josephholsten** [comment]
@shortdudey123 @msaffitz how's that suggestion feel to you both?

**Comment by msaffitz** [comment]
@josephholsten 👍  looks good to me.

File: providers/configure.rb [hunk 2]
```diff
@@ -19,10 +19,19 @@
...
+  sysconfig
   configure
   new_resource.updated_by_last_action(true)
 end
 
+def sysconfig
+  hugepage_file = "/sys/kernel/mm/transparent_hugepage/enabled"
+  file hugepage_file do
+    content 'never'
+    only_if { ::File.exists?(hugepage_file) && File.read(hugepage_file).chomp == 'never' }
```
Comment (devsibwarra) [inline providers/configure.rb]: On a new Ubuntu Xenial VM, the output of this read command looks like
```
irb(main):001:0> File.read('/sys/kernel/mm/transparent_hugepage/enabled').chomp
=> "always [madvise] never"
```

It feels like there should be some kind of string parsing to determine which THP setting is active
```
irb(main):016:0> File.read('/sys/kernel/mm/transparent_hugepage/enabled').match(/^.*\[(.*)\]/).captures.first
=> "madvise"
```

**Comment by Xorima** [comment]
Hi
We have just adopted this cookbook and are aiming to rebaseline it, as part of this we are closing down all prs older than 1 year.
If you still need this change please open a new PR and we will assess this as part of our rebaseline
Thanks for your understanding
Sous-Chefs

**Pull request closed without merging**